### PR TITLE
Add annotations

### PR DESCRIPTION
**Pull request by Mingun**
Runtime changes for https://github.com/kaitai-io/kaitai_struct_compiler/pull/191

### CODE REVIEW

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Annotation, that applied to fields, getters or setters that represents field
+ * from {@code seq} language element.
+ *
+ * @author Mingun
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface Field {
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: Please don't call it "field" — let's strive to have consitent terminology — how about "SeqAttr" or "SeqAttribute"?

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * Original identifier ({@code id} key) from {@code ksy} file.
+     *
+     * @return Identifier, that can differ from field name, if it clash with
+     *         Java reserved words. Empty string, if field was unnamed
+     */
+    String id();
+    /**
+     * Index of field in sequence of fields in the type
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: "Index of an attribute in sequence of attributes in the type."?

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @return 0-based index of field in {@code seq} language element
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: "0-based index of an attribute in `seq` KSY element"?

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    int index();
+    /**
+     * Documentation string, that attached to the field in {@code doc} language element.
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: Style/grammar: "Documentation string attached to the attribute, specified in `doc` KSY element"?

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @return Documentation string for field. If documentation is missed, returns empty string
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: "Documentation string for an attribute. If documentation is missing, returns empty string."?

File: src/main/java/io/kaitai/struct/annotations/Field.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation, that applied to fields, getters or setters that represents field
+ * from {@code seq} language element.
+ *
+ * @author Mingun
```
Comment (GreyCat) [inline src/main/java/io/kaitai/struct/annotations/Field.java]: We don't really use `@author` tag in javadocs. It's always possible to see that from version control anyway.

**Comment by Mingun** [comment]
One minor note: how can I build my branches? http://doc.kaitai.io/developers.html recommends checkout https://github.com/kaitai-io/kaitai_struct, but there will not be my changes, because they are in my fork of submodule projects (https://github.com/Mingun/kaitai_struct_java_runtime and https://github.com/Mingun/kaitai_struct_compiler). Also, how to create a stage build and run tests?

File: src/main/java/io/kaitai/struct/annotations/Field.java (hunk 1 above)
Comment (Mingun) [reply src/main/java/io/kaitai/struct/annotations/Field.java]: Ok, I choose `SeqItem`

**Comment by GreyCat** [comment]
> One minor note: how can I build my branches? http://doc.kaitai.io/developers.html recommends checkout https://github.com/kaitai-io/kaitai_struct, but there will not be my changes, because they are in my fork of submodule projects (https://github.com/Mingun/kaitai_struct_java_runtime and https://github.com/Mingun/kaitai_struct_compiler). Also, how to create a stage build and run tests?

"kaitai_struct", being what we call "an umbrella project", is essentially just an empty git repo which is used to bring all other repos in some pre-set directory layout to your machine.

You can easily clone https://github.com/kaitai-io/kaitai_struct, and then replace contents of `compiler/` with https://github.com/Mingun/kaitai_struct_compiler and `runtime/java/` with https://github.com/Mingun/kaitai_struct_java_runtime. That will give you the very same environment, but with your changes.

Once that's done, basically what you do is:

* `tests/build-compiler` to build the compiler
* `tests/build-formats` to convert all test .ksy formats into all target languages (including Java), which will give you `tests/compiled/java`
* `tests/run-java` will run all the tests for Java; results will be in `tests/test_out/java`

**Comment by Mingun** [comment]
Thank you for your explanations! Later I systemize my experience and send a PR to `doc` repository with instructions, how to work on project.

**Comment by Mingun** [comment]
So what the status of this?

File: src/main/java/io/kaitai/struct/annotations/Field.java (hunk 1 above)
Comment (GreyCat) [reply src/main/java/io/kaitai/struct/annotations/Field.java]: Ok, I take it as a challenge: we currently don't have formal straightforward way to name entities, so let's formalize it: https://github.com/kaitai-io/kaitai_struct/issues/689

**Comment by GreyCat** [comment]
Overall, it seems solid and useful. However, the devil is in the details:

* Documentation strings need major overhaul for grammar and brevity.
* We might want to review the terminology.

For now, let's go ahead with what we have. I'll try to formalize the terms in future.

**Comment by Mingun** [comment]
Thanks!

**Pull request merged**